autogluon/utils/tabular/features/feature_types_metadata.py

```diff
@@ -0,0 +1,54 @@
+import logging
+from collections import defaultdict
+
+logger = logging.getLogger(__name__)
+
+
+class FeatureTypesMetadata:
+    """
+    Contains feature type metadata information such as type family groups (feature_types_raw) and special feature type groups (feature_types_special)
+
+    feature_types_raw is the dictionary computed as output to :function:`autogluon.utils.tabular.features.utils.get_type_family_groups_df`
+    feature_types_special is an optional dictionary to communicate special properties of features to downstream models that have special handling functionality for those feature types.
+        As an example, feature_types_special might contain a key 'nlp_ngram' indicating that the list of values are all features which were generated from a nlp vectorizer and represent ngrams.
+        A downstream model such as a K-Nearest-Neighbor model could then check if 'nlp_ngram' is present in feature_types_special and drop those features if present, to speed up training and inference time.
+    """
```

Comment:
Should we list all the possible feature_types here?
Otherwise it is not easy for contributors to add new models (and they may forget to handle some feature-types).

Reply:
feature types metadata isn't made to be an authority on what the dtypes mean or what ones are valid. That is between the feature generator and the models, the metadata is just the generic link between them.

Added TODO to AutoMLFeatureGenerator to do this.